framework/platform/bpl/include/bpl/bpl_db.h

```diff
@@ -0,0 +1,97 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BPL_DB_H_
+#define _BPL_DB_H_
+
+#include "bpl.h"
+#include "bpl_err.h"
+#include <stdint.h>
+#include <string>
+#include <unordered_map>
+
+namespace beerocks {
+namespace bpl {
+
+/****************************************************************************/
+/******************************* Definitions ********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************* Structures *********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************** Functions *********************************/
+/****************************************************************************/
+
+/**
+ * @brief Add parameter to the presistent DB
+ * 
+ * @param path name of the requested section.
+ * @param object full name of the parameter.
+ * @param val Initial value of the parameter.
+ * @return true on success, false otherwise.
+ */
+bool db_add_param(const std::string &path, const std::string &object, const std::string &val);
+bool db_add_param(const std::string &path, const std::string &object, const int &val);
```

Comment:
There is no point passing `int` to a function as a const reference.  We pass parameters as const reference only if it's size is or might be greater than sizeof(int), which in our case is 4 bytes.

Reply:
This is no longer relevant, but I understood the point, thanks.